biz.aQute.launcher/src/aQute/launcher/plugin/ProjectLauncherImpl.java

```diff
@@ -260,6 +261,11 @@ public Jar executable() throws Exception {
 					runpathShas.add(sha + ";name=\"" + file.getName() + "\"");
 				} else {
 					String newPath = "jar/" + file.getName();
+					String[] parts = Strings.extension(file.getName());
```

Comment:
`Strings.extension` can return `null` if the file name has no `.` in it. Not likely but possible. The code should handle this eventuality to avoid future NPE bug reports.